web/html/src/manager/shared/websocket/useWebSocket.ts

```diff
@@ -0,0 +1,59 @@
+import * as React from "react";
+
+export function useWebSocket(
+  errors: Array<string>,
+  setErrors: Function,
+  property: string,
+  callback: (value: any) => void
+) {
+  const [webSocketErr, setWebSocketErr] = React.useState(false);
+  const [pageUnloading, setPageUnloading] = React.useState(false);
+
+  let websocket: any = {};
```

Comment:
Would typing this as `let websocket: WebSocket | undefined = undefined` work?

Reply:
Also, I'm not sure whether this variable is necessary. If I'm not mistaken, `ws` should still be in scope for the hook cleanup call in which case this should be redundant?

Reply:
right, works also without storing the `ws` outside the `useEffect()` thanks!